### DISCUSSION TITLE
Use custom grid

### DISCUSSION THREAD

**Discussion by lebeand** [pymor/pymor#1248] [Q&A] [answered]
How can I specify my own grid for the simulation?
I saw you can import a gmsh-file. But it is also possible to import the format from the PDE Toolbox from Matlab?

Can I import a 3D domain? The discretization toolkit doesn't support 3D domains.

**Comment by renefritze** [comment, marked as answer]
You are correct in that the builtin discretization can only do 1D and 2D (by design which probably should be mentioned explicitly in the tutorial/docs). If you have a 3D grid you would have to use one the PDE solvers for which pyMOR bindings exist (or [create bindings for a custom solver](https://github.com/pymor/pymor/blob/2020.2.x/src/pymordemos/fenics_nonlinear.py)). That's currently FeNiCs, NgSolve and Deal.II. At the moment we don't have a tutorial using any of these IIRC, but if you take a look at the demos you will find FeNiCs and NgSolve used. [This one for example](https://github.com/pymor/pymor/blob/2020.2.x/src/pymordemos/fenics_nonlinear.py#L82) does 3D + FeNiCs
